R/pagination_helpers.R

```diff
@@ -88,3 +88,40 @@ fetch_until <- function(fn, .until, ...) {
 
   full_results
 }
+
+#' Retrieve items from a paginated endpoint matching keys
+#' @param fn function, the API function to be called (usually a list endpoint)
+#' @param keys list of key value pairs.
+#' @return Items from \code{fn} that match all key value pairs in \code{keys}.
+#' @param ... Arguments passed to \code{fn}.
+#' @examples
+#' \dontrun{
+#' find(credentials_list, list(username = 'username', name = 'cred_name')))
+#' find(templates_list_scripts, list(name = 'template name'))
+#'
+#' }
+#' @export
+find <- function(fn, keys, ...) {
```

Comment:
if I understand the code correctly, should this be "find_all" instead of "find"?  I.e. you'll loop over all pages regardless if you find a match?